Made history longer

Wuerfel21 opened the pull request:
It was too short, i always needed to retype long commands

Wuerfel21 committed:
Made history longer

It was too short, i always needed to retype long commands

Wuerfel21 commented on src/main/resources/assets/opencomputers/loot/OpenOS/bin/sh.lua:
```diff
@@ -165,7 +165,7 @@ if #args == 0 and (io.input() == io.stdin or options.i) and not options.c then
         return -- eof
       end
-      while #history > 10 do
+      while #history > 20 do
```
derp, 2 times more history

fnuecke commented:
That's still kinda arbitrary, and the low limit was intended to avoid (relatively) non-obvious memory consumption. I think the better approach would be to add a `HISTSIZE` env var (defaulting to 10) and use that there. Then you can just `set HISTSIZE=20` in the shell and be happy ;-)

Wuerfel21 commented:
Please hold the line, your request is beeing processed...

Wuerfel21 committed:
Made Histrory Length a EnvVar

Well, there is it!

Wuerfel21 commented:
Well, I done it(hopefully right)

Wuerfel21 commented on src/main/resources/assets/opencomputers/loot/OpenOS/bin/sh.lua:
```diff
@@ -7,6 +7,14 @@ local shell = require("shell")
 local term = require("term")
 local text = require("text")
 
+local function getHistSize()
```
This Function does stuff(really?) to either return HISTSIZE or 10

fnuecke commented on src/main/resources/assets/opencomputers/loot/OpenOS/boot/01_os.lua:
```diff
@@ -16,6 +16,7 @@ local env = {
   SHELL="/bin/sh",
   TMP="/tmp", -- Depricated
   TMPDIR="/tmp"
```
Missing a comma there. Test your stuff! :O

fnuecke replied:
Also, probably better make it `HISTSIZE="10"`, for consistency (don't mix in non-string types into env-vars).

fnuecke commented on src/main/resources/assets/opencomputers/loot/OpenOS/bin/sh.lua:
```diff
@@ -165,7 +173,7 @@ if #args == 0 and (io.input() == io.stdin or options.i) and not options.c then
         return -- eof
       end
-      while #history > 10 do
+      while #history > getHistSize() do
```
Maybe just a `(tonumber(os.getenv("HISTSIZE")) or 10)` here, do away with the function?

Wuerfel21 commented:
ok

Wuerfel21 committed:
fixed random stuff

well, title states.
and there is some random binary for you:
11000011 01011101
10101110 11001111
00100011 00110110
01011000 01110101
11101111 11010100
11100101 10010011
00011001 11100101
10000101 10100000
11011001 01110110 10000111 00111000 00010000 10011001 10100111 10111100
10000101 00010000 01010010 10111101 10011101 00010101 10010001 00001101
10110100 10101111
00110001 11010101
01011010 11000000
00000100 10101011
01011111 11001111

fnuecke commented:
Thanks :)

Wuerfel21 commented:
Phew... No more typeing madness...
